lambda/index.ts

```diff
@@ -26,12 +29,122 @@ type SocialAccountType =
+            }
+        });
+        if (response.status === 200) {
+            return true;
+        } else {
+            console.error('Failed to add domain to denylist. API returned error:', response.data);
+            return false;
+        }
+    } catch (error) {
+        console.error('Unexpected error while adding domain to denylist:', error);
+        return false;
+    }
+}
+    
+export async function checkPrivyDenylist(email: string): Promise<boolean> {
+    const emailDomain = email.split('@')[1];
+    try {
+        const response = await axios.get(`https://auth.privy.io/api/v1/apps/${PRIVY_APP_ID}/denylist`, { // Initial request to get the denylist, no cursor query parameter
+            auth: {
+                username: PRIVY_APP_ID,
+                password: PRIVY_APP_SECRET
+            },
+            headers: {
+                'privy-app-id': PRIVY_APP_ID
+            }
+        });
+
+        if (!response.data.data) {
+            return false;
+        }
+
+        const exists = response.data.data.some((dataObject: { value: string }) => dataObject.value === emailDomain);
+        if (exists) {
+            return true;
+        }
+
+        let cursor = response.data?.next_cursor;
+        while (cursor) {
+            const nextResponse = await axios.get(`https://auth.privy.io/api/v1/apps/${PRIVY_APP_ID}/denylist?cursor=${cursor}`, {
+                auth: {
+                    username: PRIVY_APP_ID,
+                    password: PRIVY_APP_SECRET
+                },
+                headers: {
+                    'privy-app-id': PRIVY_APP_ID
+                }
+            });
+
+            const exists = nextResponse.data.data.some((dataObject: { value: string }) => dataObject.value === emailDomain);
+            if (exists) {
+                return true;
+            }
+            cursor = nextResponse.data.next_cursor;
+        }
+        return false;
+    } catch (error) {
+        console.error('Error checking email disposability:', error);
+        return false;
+    }
+}
+
+export async function tryUserCheck(email: string): Promise<boolean> {
+    const USERCHECK_API_KEY = process.env.USERCHECK_API_KEY as string;
+    if (!USERCHECK_API_KEY) {
+        return false;
+    }
+    try {
+        const response = await axios.get(`https://api.usercheck.com/domain/${email.split('@')[1]}`, {
+            headers: {
+                'Authorization': `Bearer ${USERCHECK_API_KEY}`
+            }
+        });
+        if (response.status === 200) {
+            return response.data.disposable;
+        } else if (response.status === 400) {
+            console.error('Error domain is invalid:', response.data);
+            return false;
+        } else { // Should be only status 429
+            console.error('Error too many requests:', response.data);
+            return false;
+        }
+    } catch (error) {
+        console.error('Error checking email disposability:', error);
+        return false;
+    }
+}
```

Comment:
Looks good, @mikeredmond! we should consider creating a generic API handler and centralised route config, so we can reuse the API call logic and simplify adding new integrations. also, so it's easy to switch endpoints/constants